### PR TITLE
observability: add two more timers

### PR DESCRIPTION
**Pull request by drahnr**
Part of series to improve observability in availability related subsystems

Ref https://github.com/paritytech/polkadot/issues/5106

### CODE REVIEW

File: node/network/availability-recovery/src/metrics.rs [hunk 1]
```diff
@@ -122,7 +144,21 @@ impl metrics::Metrics for Metrics {
 			time_chunk_request: prometheus::register(
 				prometheus::Histogram::with_opts(prometheus::HistogramOpts::new(
 					"polkadot_parachain_availability_recovery_time_chunk_request",
-					"Time spent waiting for a response to a chunk request",
+					"Time spent waiting f requesting  for a response to a chunk requestor a response to a chunk request",
```
Comment (eskimor) [inline node/network/availability-recovery/src/metrics.rs:147]: That does not look right.

**Review by eskimor — APPROVED** [review]

File: node/network/availability-recovery/src/metrics.rs (hunk 1 above)
Comment (drahnr) [inline node/network/availability-recovery/src/metrics.rs:147]:
```suggestion
					"Time spent waiting for a response to a chunk request",
```

**Comment by drahnr** [comment]
bot merge

File: node/network/availability-recovery/src/metrics.rs [hunk 2]
```diff
@@ -42,8 +42,15 @@ struct MetricsInner {
...
 	chunk_requests_finished: CounterVec<U64>,
+
 	/// The duration of request to response.
-	time_chunk_request: prometheus::Histogram,
+	time_chunk_request: Histogram,
+
+	/// Time the pure recovery and verification duration.
+	time_erasure_recovery: Histogram,
+
+	/// Time the time from the first request, to having a sufficient number to recover.
```
Comment (vstakhov) [inline node/network/availability-recovery/src/metrics.rs:52]: This comment looks wrong, I think:

```
/// The time from the first request to the time when we have a sufficient number of chunks to recover
```

File: node/network/availability-recovery/src/metrics.rs [hunk 3]
```diff
@@ -42,8 +42,15 @@ struct MetricsInner {
 	/// - `invalid` ... Chunk was received, but not valid.
 	/// - `success`
 	chunk_requests_finished: CounterVec<U64>,
+
 	/// The duration of request to response.
-	time_chunk_request: prometheus::Histogram,
+	time_chunk_request: Histogram,
+
+	/// Time the pure recovery and verification duration.
```
Comment (vstakhov) [inline node/network/availability-recovery/src/metrics.rs:49]: `s/Time the/Duration between/` probably?

**Review by vstakhov — APPROVED** [review]

**Comment by drahnr** [comment]
bot merge

**Comment by drahnr** [comment]
bot merge

**Pull request merged**